packages/compass-connections-navigation/src/connections-navigation-tree.tsx

```diff
@@ -277,14 +278,24 @@ const NavigationItem = memo<{
     items,
     isSingleConnection,
     isReadOnly,
-    activeNamespace,
+    activeWorkspace,
     currentTabbable,
     onConnectionExpand,
     onConnectionSelect,
     onDatabaseExpand,
     onNamespaceAction,
   } = data;
 
+  const {
+    connectionId: activeConnectionId,
+    namespace: activeNamespace,
+    type: activeWorkspaceType,
+  } = activeWorkspace as {
+    connectionId?: string;
+    namespace?: string;
+    type?: string;
+  };
```

Comment:
I have started out with checks for the type, but in the end I felt like that makes the code unnecessarily dependent on the workspaces. For example I could blacklist `My Queries` because it's the only one that doesn't have connectionId, but later we add `My Tests` and now this doesn't compile. Meanwhile for the logic we only care if the workspace is connection scoped or not.

Reply:
I'll give it another try and perhaps I can only do assertions where the type doesn't matter and still rely on the types where the type is checked anyway

Reply:
updated - now the casting is restricted to the autoexpanding bit